pkg/e2e/infra/utils.go

```diff
@@ -18,11 +18,17 @@ import (
 	mapiv1beta1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
 	controllernode "github.com/openshift/cluster-api/pkg/controller/node"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const nodeWorkerRoleLabel = "node-role.kubernetes.io/worker"
+const (
+	nodeWorkerRoleLabel        = "node-role.kubernetes.io/worker"
+	deprecatedMachineRoleLabel = "sigs.k8s.io/cluster-api-machine-role"
+	machineRoleLabel           = "machine.openshift.io/cluster-api-machine-role"
```

Comment:
This is exactly the whole point. If this does not work we have a problem since our components rely on it e.g aws-actuator. We definitely want to keep and test this so we know when it's not working

Reply:
Openshift specific labels disallow to run the test suite over pure Kubernetes. Are you aware of that?

Reply:
Rather than hard code this, pass the `labels []string` in.

Reply:
`getWorkerMachineSetList` is intentionally not allowing the label as a parameter to satisfy its name.
@ingvagabund Openshift specific must be tested here, there's machineSet with zero replicas that you can't not get from nodes